subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -0,0 +1,440 @@
+import {
+  Address,
+  BigInt,
+  BigDecimal,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import { bigIntToBigDecimal } from "../utils/numbers";
+import { getOrCreateProtocol } from "../entities/protocol";
+import { getOrCreatePool } from "../entities/pool";
+import {
+  FinancialsDailySnapshot,
+  Minipool,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+} from "../../generated/schema";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  ETH_ADDRESS,
+  RETH_ADDRESS,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+  RPL_ADDRESS,
+  BIGDECIMAL_HALF,
+} from "../utils/constants";
+import { getOrCreateRewardToken, getOrCreateToken } from "../entities/token";
+import { getOrCreateMinipool } from "../entities/Minipool";
+
+const PROTOCOL_ID = RETH_ADDRESS;
+
+export function updateMinipoolTvlandRevenue(
+  block: ethereum.Block,
+  stakingAmount: BigInt,
+  slashingAmount: BigInt,
+  totalRewardEth: BigInt,
+  miniPoolAddress: string
+): void {
+  const Minipool = getOrCreateMinipool(
+    block.number,
+    block.timestamp,
+    miniPoolAddress
+  );
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  Minipool.inputTokenBalances = [
+    Minipool.inputTokenBalances[0].plus(stakingAmount),
+  ];
+
+  Minipool.slashAmount = Minipool.slashAmount.plus(slashingAmount);
+
+  var computebalance: BigInt = BIGINT_ZERO;
+  let balance = Minipool.inputTokenBalances[0].minus(Minipool.slashAmount);
+  if (BigInt.compare(balance, BIGINT_ZERO) > 0) {
+    computebalance = balance;
+  }
+  Minipool.totalValueLockedUSD = bigIntToBigDecimal(computebalance).times(
+    getOrCreateToken(Address.fromString(ETH_ADDRESS), block.number)
+      .lastPriceUSD!
+  );
+
+  pool.miniPoolTotalValueLocked = pool.miniPoolTotalValueLocked.plus(
+    Minipool.inputTokenBalances[0]
+  );
```

Comment:
Is this correct? Shouldn't we only add the delta?

Reply:
We're adding each minipool value into pool value as the minipool is initialized. so this is correct.